<launch>
  <arg name="output" default="log"/>
  <arg name="standalone" default="false"/>

  <!-- World setup -->
  <include if="$(arg standalone)" file="$(find agimus_demos)/launch/tiago_deburring_world_setup.launch">
  </include>

  <!-- Setup vision -->
  <group ns="vision">
    <node pkg="agimus_vision" type="tracker_object"
        output="$(arg output)"
        name="tracker"
        >
        <param name="imageTopic"      value="/xtion/rgb/image_raw" />
        <param name="cameraInfoTopic" value="/xtion/rgb/camera_info" />
        <param name="debugDisplay"    value="false" />
        <param name="broadcastTopic"  value="true" />
        <param name="broadcastTf"     value="true" />
        <param name="broadcastTfPostfix" value="_tf" />
        <param name="objectType" value="apriltag" />
        <param name="cameraFrame" value="xtion_optical_frame" />
    </node>

    <!-- Part -->
    <node pkg="agimus_demos" type="group_of_tags.py" name="part_tags" args="
        --child-frame-format 'part/tag36_11_{:0>5d}'
        --measurement-parent-frame xtion_optical_frame
        --group-frame part/base_link
        --no-description
        --size 0.0573
        --tag 100
        --tag 101
        --tag 102
        --tag 103
        "
        output="$(arg output)"
        />
  </group>
</launch>
